sympy/codegen/ast.py

```diff
@@ -399,3 +448,515 @@ def body(self):
         Must be an iterable object or CodeBlock.
         """
         return self._args[2]
+
+
+class Token(Basic):
+    """ Similar to Symbol, but takes no assumptions.
+
+    Defining fields are set in __slots__.
+    """
+
+    __slots__ = []
+
+    def __new__(cls, *args, **kwargs):
+        if len(args) == 1 and not kwargs and isinstance(args[0], cls):
+            return args[0]
+        args = args + tuple([kwargs[k] for k in cls.__slots__[len(args):]])
+        return Type.__xnew_cached_(cls, *args)
+
+    def __new_stage2__(cls, *args):
+        obj = Basic.__new__(cls)
+        for attr, arg in zip(cls.__slots__, args):
+            setattr(obj, attr, arg)
+        return obj
+
+    __xnew_cached_ = staticmethod(cacheit(__new_stage2__))
```

Comment:
I think we don't need all this stage2 junk here (or even caching, for that matter). As far as I can tell, it's used in Symbol so that symbols are cached based on derived assumptions (as opposed to the assumptions passed in by the user).